CRM/Fastactivity/BAO/Activity.php

```diff
@@ -214,6 +219,27 @@ public static function whereClause(&$params, $sortBy = TRUE, $excludeHidden = TR
     if (!empty($activity_type_id)) {
       $clauses[] = "activity.activity_type_id IN ( " . $activity_type_id . " ) ";
```

Comment:
This allows for SQL injection - use this pattern to encode the value for SQL:

```php
$clauses[] = "activity.activity_type_id IN (%2) ";
$params[2] = [$activity_type_id, 'CommaSeparatedIntegers'];
```

This issue is repeated a few times (including in old code) - I'll mark it with "SQLi".